x-pack/platform/plugins/shared/streams/server/lib/rules/esql/register.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AlertInstanceContext, RuleTypeState } from '@kbn/alerting-plugin/server';
+import { alertFieldMap } from '@kbn/alerts-as-data-utils';
+import { DEFAULT_APP_CATEGORIES } from '@kbn/core/server';
+import { LicenseType } from '@kbn/licensing-plugin/server';
+import { ESQL_RULE_TYPE_ID } from '@kbn/rule-data-utils';
+import { PersistenceAlertType } from '@kbn/rule-registry-plugin/server';
+import {
+  STREAMS_FEATURE_ID,
+  STREAMS_RULE_REGISTRATION_CONTEXT,
+} from '../../../../common/constants';
+import { getRuleExecutor } from './executor';
+import { EsqlRuleParams, esqlRuleParams } from './types';
+
+export function esqlRuleType(): PersistenceAlertType<
+  EsqlRuleParams,
+  RuleTypeState,
+  AlertInstanceContext,
+  'default'
+> {
+  return {
+    id: ESQL_RULE_TYPE_ID,
+    name: 'ES|QL Rule',
+    validate: {
+      params: {
+        validate: (object: unknown) => {
+          return esqlRuleParams.parse(object);
+        },
+      },
+    },
+    schemas: {
+      params: { type: 'zod', schema: esqlRuleParams },
+    },
+    defaultActionGroupId: 'default',
+    actionGroups: [
+      {
+        id: 'default',
+        name: 'Default',
+      },
+    ],
+    minimumLicenseRequired: 'enterprise' as LicenseType,
+    category: DEFAULT_APP_CATEGORIES.observability.id,
+    producer: STREAMS_FEATURE_ID,
+    solution: 'observability',
+    isExportable: false,
+    actionVariables: {},
+    executor: getRuleExecutor,
+    autoRecoverAlerts: false,
+    alerts: {
+      context: STREAMS_RULE_REGISTRATION_CONTEXT,
+      mappings: { dynamic: false, fieldMap: { ...alertFieldMap } },
+      shouldWrite: false,
+      isSpaceAware: true,
```

Comment:
`isSpaceAware: true` will create a new set of resources for every space a rule runs in, so `.alerts-streams.alerts-default`, `.alerts-streams.alerts-space1`, `.alerts-streams.alerts-space2`. Just checking that that is the desired outcome? For most of the o11y rules, it is sufficient that the space ID is written to each alert document and the alert UIs from each space only show the alerts with the correct space ID, regardless of what index it's written to.